lib/compress.js

```diff
@@ -2251,7 +2251,8 @@ merge(Compressor.prototype, {
                             dirs.push(node);
                             return make_node(AST_EmptyStatement, node);
                         }
-                        if (node instanceof AST_Defun && hoist_funs) {
+                        if (node instanceof AST_Defun && hoist_funs
+                            && (tt.parent() === self || !compressor.has_directive("use strict"))) {
```

Comment:
`instanceof` is more than 10X slower than a boolean variable check, so could you put the `hoist_funs` condition first?

I appreciate this would only help the case when `hoist_funs` is disabled, but it also makes the code clearer.